crates/sparrow-instructions/src/evaluators/map/get.rs

```diff
@@ -0,0 +1,123 @@
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+use arrow::array::{Array, ArrayRef, PrimitiveArray, StringArray};
+use arrow::datatypes::{ArrowPrimitiveType, DataType};
+use itertools::Itertools;
+use owning_ref::ArcRef;
+use sparrow_arrow::downcast::{downcast_primitive_array, downcast_string_array};
+
+use sparrow_plan::ValueRef;
+
+use crate::{Evaluator, EvaluatorFactory, StaticInfo};
+
+/// Evaluator for `get` on maps.
+#[derive(Debug)]
+pub(in crate::evaluators) struct GetStringToPrimitiveEvaluator<T>
```

Comment:
I don't think this should need to specialize on the value type, if it uses the `take` kernel to create the values array. That would allow this to specialize only on the key type. And we can use `eq_dyn` to not specialize on the key type either.

Reply:
Ah nice, will try that. Was going to add the vectorized comparison before making this non-draft, so will see if that works

Reply:
The trouble with the vectorized comparison is that `map.keys()` flattens all entries, resulting in an array of a different length than the `key_input`. As you mentioned, the new version of arrow that supports the new `datum` trait may be of help to clean this up. For now, I'd like to merge this less efficient implementation in though